sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -1781,6 +1781,32 @@ abstract class CSVSuite
     }
   }
 
+  test("SPARK-49016: Queries from raw CSV files are disallowed when the referenced columns only" +
+    " include the internal corrupt record column") {
+    val schema = new StructType()
+      .add("a", IntegerType)
+      .add("b", DateType)
+      .add("_corrupt_record", StringType)
+
+    // negative cases
+    checkError(
+      exception = intercept[AnalysisException] {
+        spark.read.schema(schema).csv(testFile(valueMalformedFile))
+          .select("_corrupt_record").collect()
```

Comment:
what was the behavior before this PR?

Reply:
The previous behavior looked like this:
![image](https://github.com/user-attachments/assets/b57e04df-69fd-4eb1-a9fd-4bea93df2c91)

Reply:
@wayneguow  the query itself makes sense, but the results "NULL" are wrong. Blocking this looks incorrect to me. 

Could you revert it ?

Reply:
@gatorsmile Sorry, I may not understand what you mean, you said that the results "NULL" are wrong, but it is the previous behavior before this PR, so why we need to revert this?

Reply:
The definition of "corrupted record" has been unclear for a while, as it depends on the column being read. This change itself is a breaking change as it introduced a new error.

Reply:
Hmmm, from this point of view, this PR can be reverted. I respect your advices and you have more experience about this. If it's convenient for you, you can help to revert it. Thank you.